### ISSUE TITLE
A warning is printed to the Command Line during Prebuilt SDK build on certain PCs

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#6241]
**Describe the bug**
Configuring and building the engine following the Prebuilt SDK instruction set causes a warning to display on certain PCs. It should be noted that the build itself goes through successfully, and the engine does not have any apparent issues caused by this.

Warning displayed:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(5
13,5): warning MSB8028: The intermediate directory (x64\profile\) contains files shared from another project (LmbrCentr
al.API.vcxproj).  This can lead to incorrect clean and rebuild behavior. [C:\Builds\o3de\build\windows_vs2019\External\
LmbrCentral-dfc60e35\Code\LmbrCentral.Mocks.vcxproj]
```

**Steps to reproduce**
1. Configure and build the engine following the Prebuilt SDK instruction set.

**Expected behavior**
Build goes through without any issues.

**Actual behavior**
A warning is displayed during the build.

**Command Line log and dxdiags**
Command Line log:
```
C:\Builds\o3de>cmake --build build/windows_vs2019 --target INSTALL --config profile -- /m
Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/Builds/o3de/Code/Framework/AzCore/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/AtomLyIntegration/CommonFeatures/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/Atom/Bootstrap/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/Atom/Asset/ImageProcessingAtom/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Code/Editor/Plugins/EditorCommon/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Code/Editor/CMakeLists.txt
  unity_28_cxx.cxx
  unity_25_cxx.cxx
  unity_24_cxx.cxx
  unity_27_cxx.cxx
  unity_26_cxx.cxx
  unity_22_cxx.cxx
  unity_29_cxx.cxx
  unity_21_cxx.cxx
  unity_18_cxx.cxx
  unity_20_cxx.cxx
  unity_23_cxx.cxx
  unity_19_cxx.cxx
  unity_15_cxx.cxx
  unity_16_cxx.cxx
  unity_17_cxx.cxx
  unity_14_cxx.cxx
  unity_13_cxx.cxx
  unity_12_cxx.cxx
  unity_11_cxx.cxx
  unity_10_cxx.cxx
  unity_9_cxx.cxx
  unity_8_cxx.cxx
  unity_7_cxx.cxx
  unity_6_cxx.cxx
  unity_5_cxx.cxx
  unity_4_cxx.cxx
  unity_3_cxx.cxx
  unity_2_cxx.cxx
  unity_1_cxx.cxx
  unity_0_cxx.cxx
  HeapSchema.cpp
  Building Custom Rule C:/Builds/o3de/Code/Legacy/CryCommon/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/GameStateSamples/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/GradientSignal/Code/CMakeLists.txt
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(5
13,5): warning MSB8028: The intermediate directory (x64\profile\) contains files shared from another project (LmbrCentr
al.API.vcxproj).  This can lead to incorrect clean and rebuild behavior. [C:\Builds\o3de\build\windows_vs2019\External\
LmbrCentral-dfc60e35\Code\LmbrCentral.Mocks.vcxproj]
  Building Custom Rule C:/Builds/o3de/Gems/LmbrCentral/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/LmbrCentral/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/Presence/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/PhysX/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/ScriptCanvas/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/Terrain/Code/CMakeLists.txt
  AzCore.vcxproj -> C:\Builds\o3de\build\windows_vs2019\lib\profile\AzCore.lib
  Building Custom Rule C:/Builds/o3de/Gems/GameState/Code/CMakeLists.txt
  Copying CustomAssetExample runtime dependencies to output...
  Running AutoGen for AzNetworking
  Building Custom Rule C:/Builds/o3de/Code/Framework/AtomCore/CMakeLists.txt
  Copying ImGui.imguilib runtime dependencies to output...
  Building Custom Rule C:/Builds/o3de/Gems/PhysX/Code/NumericalMethods/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Code/Framework/AzTest/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/ScriptedEntityTweener/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Code/Tools/CrashHandler/Support/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/CustomAssetExample/Code/CMakeLists.txt
  Building Custom Rule C:/Builds/o3de/Gems/ImGui/Code/CMakeLists.txt
  Copying GraphCanvas runtime dependencies to output...
  Building Custom Rule C:/Builds/o3de/Gems/GraphCanvas/Code/CMakeLists.txt
  Copying SceneProcessing runtime dependencies to output...
  Building Custom Rule C:/Builds/o3de/Gems/SceneProcessing/Code/CMakeLists.txt
```
Dxdiag files:
PCs which reproduced the issue:
[DxDiag_GPC-464.txt](https://github.com/o3de/o3de/files/7674809/DxDiag_GPC-464.txt)
[DxDiag_GPC-602.txt](https://github.com/o3de/o3de/files/7674795/DxDiag_GPC-602.txt)
[DxDiag_GPC-610.txt](https://github.com/o3de/o3de/files/7674816/DxDiag_GPC-610.txt)

PC which did not reproduce the issue:
[DxDiag-GPC-462.txt](https://github.com/o3de/o3de/files/7674790/DxDiag-GPC-462.txt)

**Found in Branch**
Development (a8ef23e)

**Desktop**
Device: PC
OS: Windows
Version 10
CPU AMD Ryzen 5 3600 6-Core Processor, 3600 MHz, 6 Core(s), 12 Logical Processor(s)
GPU NVIDIA GeForce RTX 2060 SUPER
Memory 16GB

**Comment by amznestebanpapp** [comment]
I debugged this and is a bug in CMake, logged the issue to CMake: https://gitlab.kitware.com/cmake/cmake/-/issues/23043
The warning is benign, I will take a look to see if we can workaround it by injecting a props file in these targets

**Comment by amznestebanpapp** [comment]
As a temporal solution, I am going to move the test targets into the Test folder in LmbrCentral, that will make the Mocks target to be in a different folder than the Headers folder, removing this warning

**Comment by amznestebanpapp** [comment]
yeah, this will be a nightmare to support... besides some scripts, every place in cmake that deals with paths, has to account for that...
In CMake, a space is also a list separator in the invocation of a method, so something like:
```
set(some_path <some path with spaces>)
file(WRITE ${some_path} CONTENTS ...)
```
has to be written:
```
set(some_path "<some path with spaces>")
file(WRITE "${some_path}" CONTENTS ...)
```
I did a couple of searches and we have this all over the place, some places are passing quotes properly, but most places are not. It will require a full pass to validate it. We would also have to make sure that it doesnt regress since is very delicate.
Besides that, we would have to check that works for the tools, e.g. when passing arguments to commands it has to escape/consider it.

Another example where other things will break, every call to `subprocess.run` in python needs to use `[]` or the cmd wrapped in `f"{}"`

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (f09055a) build. The error did not reappear when performing a clean Prebuilt SDK build today.
